package.json

```diff
@@ -9,36 +9,53 @@
   "engines": {
     "node": ">=10"
   },
+  "dependencies": {},
   "devDependencies": {
-    "@types/chai": "^4.2.6",
-    "@types/mocha": "^5.2.7",
+    "@ethersproject/providers": "^5.4.7",
+    "@nomicfoundation/hardhat-chai-matchers": "^1.0.1",
+    "@nomicfoundation/hardhat-network-helpers": "^1.0.2",
+    "@nomiclabs/hardhat-ethers": "^2.0.0",
+    "@nomiclabs/hardhat-etherscan": "^2.1.6",
+    "@typechain/ethers-v5": "^10.1.0",
+    "@typechain/hardhat": "^6.1.2",
+    "@types/chai": "^4.2.14",
+    "@types/mocha": "^9.1.1",
+    "@typescript-eslint/eslint-plugin": "^4.29.3",
+    "@typescript-eslint/parser": "^4.29.3",
     "arb-ethers-web3-bridge": "^0.7.3",
-    "chai": "^4.2.0",
-    "dotenv": "^8.2.0",
-    "ethereum-waffle": "^2.4.1",
-    "ethereumjs-util": "^6.2.0",
-    "ethers": "^4.0.45",
-    "mocha": "^6.2.2",
-    "prettier": "^1.19.1",
+    "chai": "^4.3.4",
+    "dotenv": "^10.0.0",
+    "eslint": "^7.32.0",
+    "ethereumjs-util": "^7.1.3",
+    "ethers": "^5.4.6",
+    "hardhat": "^2.10.0",
+    "hardhat-dependency-compiler": "^1.1.3",
+    "hardhat-deploy": "^0.9.0",
+    "hardhat-deploy-ethers": "^0.3.0-beta.10",
+    "hardhat-gas-reporter": "^1.0.8",
+    "mathjs": "^10.6.1",
+    "prettier": "^2.3.2",
+    "prettier-plugin-solidity": "^1.0.0-beta.17",
     "rimraf": "^3.0.0",
     "solc": "0.5.16",
-    "truffle-flattener": "^1.4.4",
-    "truffle-hdwallet-provider": "^1.0.17",
-    "ts-node": "^8.5.4"
+    "solhint": "^3.3.6",
+    "solhint-plugin-prettier": "0.0.5",
+    "solidity-coverage": "^0.7.20",
+    "ts-node": "10.7.0",
+    "typechain": "^8.1.0",
+    "typescript": "^4.5.2"
   },
   "scripts": {
     "lint": "yarn prettier ./test/*.ts --check",
     "lint:fix": "yarn prettier ./test/*.ts --write",
     "clean": "rimraf ./build/",
     "precompile": "yarn clean",
-    "compile": "waffle .waffle.json",
-    "pretest": "yarn compile",
-    "test": "mocha",
-
+    "compile": "hardhat compile",
+    "test": "hardhat compile && hardhat test",
```

Comment:
More of a preference than anything, but maybe calling the `compile` script here instead of `hardhat compile` will make the `test` script future-proof. Let's say something changes in the compilation command and we add stuff to it, maybe that stuff is also needed while testing. In that case, without the `compile` script being called in the `test` one, we would need to remember to manually update the `test` script too. Veeeery small thing, a non-blocker.

Reply:
Changed as suggested with [d6c5c9d](https://github.com/levelkdev/dxswap-core/pull/45/commits/d6c5c9df1aeb49ccb9f827ea6c7e3df2d16f6a4d)